### ISSUE TITLE
Nearest Neighbor/Integer screen scaling

### ISSUE THREAD

**Issue by EpicGamerNaeNae** [flyinghead/flycast#1759]
**Is your feature request related to a problem? Please describe.**
By default, when playing at internal resolutions smaller than the window/screen, Flycast uses linear filtering to scale the image up, which results in a blurry image.

This behavior can be changed, by changing the texture filtering option from default or linear to nearest-neighbor, but displays the textures differently than was intended.

**Describe the solution you'd like**
There should be an option for screen scaling, separate from the texture scaling option, to allow to choose between bilinear filtering, nearest neighbor, or integer scaling (displaying the game only at exact multiples of the original resolution).
The sharper output would also help when using external filtering options.

**Describe alternatives you've considered**
External software like lossless scaling can be used to achieve nearest neighbor/integer scaling, but it is a bit finicky (at 1440p you need to set it to 2.999x scaling (854x480) to look fine, since 3x (853x480) is one pixel off and makes the image look terrible), and isn't accessible to all users.

**Additional context**
Screenshots at 1440p, with internal resolution at 1x:

Default texture filtering, textures look the way they're supposed to, but the image is blurry:
![image](https://github.com/user-attachments/assets/124555b1-1c69-4def-b567-5a20f74dc654)

Nearest Neighbor texture filtering, textures look worse, but the image is sharper:
![image](https://github.com/user-attachments/assets/9a951e79-2d68-4e61-8c6e-b3a2a4c952c1)

Default texture filtering, with lossless scaling resizing the window to 854x480 (scale factor 2.999 on a 1440p display, scaled up to 1440 using nearest neighbor on paint.net). Intended look for the textures, with a sharp image:
![image](https://github.com/user-attachments/assets/634ffa9e-d806-42ff-92f2-f6ffcd1b0795)

**Comment by flyinghead** [comment]
Duplicate of #1415

**Comment by SpelzR2** [comment]
It isn't a duplicate of #1415 as texture filtering isn't the same as screen scaling.

**Comment by skullpanda86** [comment]
> It isn't a duplicate of [#1415](https://github.com/flyinghead/flycast/issues/1415) as texture filtering isn't the same as screen scaling.

Exactly, why is screen scaling filtering being confused with texture filtering?

**Comment by flyinghead** [comment]
Fixed on the `dev` branch
